lib/Doctrine/Migrations/Metadata/Storage/MetadataStorage.php

```diff
@@ -13,7 +14,12 @@ public function ensureInitialized(): void;
 
     public function getExecutedMigrations(): ExecutedMigrationsList;
 
-    public function complete(ExecutionResult $migration): void;
+    public function complete(ExecutionResult $result): void;
 
     public function reset(): void;
+
+    /**
+     * @return Query[]
+     */
+    public function getSql(ExecutionResult $result): array;
```

Comment:
Adding a method to an interface for implementors outside the package. You can use an `@method` annotation to indicate that the method _should_ be present, it will be taken into account by static analysis tools I think.

Reply:
Also, you don't need the return value to be an `array`, iterable is enough.

Reply:
> Adding a method to an interface for implementors outside the package. You can use an `@method` annotation to indicate that the method _should_ be present, it will be taken into account by static analysis tools I think.

what do you mean by "_should_ be present". This method always should be present in implementation because this method exists in Interface. MetadataStorage is interface. All methods in interfaces should be implemented even without @method annotation. Or what do you mean?

I think about getSql() method and maybe we should introduce new interface? Like MetadataSQLStorage and move getSql method there? So in DbalExecutor we will call this method only if our Storage implement this interface. What do you think?

Reply:
>      what do you mean by "should be present". 

I mean "should be present"… if the implementor wants to become compatible with the next major release of `doctrine/migrations`, where the method should be directly in the interface.

> I think about getSql() method and maybe we should introduce new interface? Like MetadataSQLStorage and move getSql method there? So in DbalExecutor we will call this method only if our Storage implement this interface. What do you think?

I think you should use `method_exists`, and then, on branch 3.6, trigger a deprecation if is not implemented. The deprecation should state that `getSql()` will become part of the interface in 4.0.0

Reply:
Yes, please do.

Reply:
done